src/core/lv_obj_class.h

```diff
@@ -59,6 +59,12 @@ typedef struct _lv_obj_class_t {
     void (*event_cb)(const struct _lv_obj_class_t * class_p,
                      struct _lv_event_t * e);  /**< Widget type specific event function*/
     void * user_data;
+#if LV_USE_CLASS_NAME
+    const char * name;
+#endif
+#if LV_USE_OBJID_BUILTIN
+    uint32_t * obj_count; /* Points to a static variable for logging object counts */
```

Comment:
Why does it need to be pointer? Can't we use a simple `uint32_t`?

Reply:
class like `lv_obj_class` is always defined as `const`, so we have to use pointer to another variable.

```c
#if LV_USE_OBJID_BUILTIN
    static uint32_t obj_count = 0;
#endif

const lv_obj_class_t lv_obj_class = {
#if LV_USE_OBJID_BUILTIN
    .obj_count = &obj_count,
#endif

#if LV_USE_CLASS_NAME
    .name = "obj",
#endif
};

```

Reply:
Doh, that's true. I still wonder if we can figure out something cleaner. 

A simple solution wold be to have a `static uint32_t id;` in the assign_id function, but this way we couldn't count the number of widgets by type, only a global ID. 

Do you think it wold be suitable as well?

Reply:
We can use dynamic array to track each type of widget and resize the array when new class type is used.

Reply:
I have pushed the patch to remove urgly `obj_count`.